src/services/notifications/index.ts

```diff
@@ -0,0 +1,27 @@
+import { LedgerTransactionType } from "@domain/ledger"
+import { NotificationsServiceError } from "@domain/notifications"
+import { User } from "@services/mongoose/schema"
+import { transactionNotification } from "./payment"
+
+export const NotificationsService = (logger: Logger): INotificationsService => {
+  const onChainTransactionReceived = async ({
+    amount,
+    walletId,
+    txId,
+  }: TransactionReceivedArgs) => {
+    try {
+      // work around to move forward before re-wrighting the whole notifications module
+      const user = await User.findOne({ _id: walletId })
+      transactionNotification({
```

Comment:
I think this should be called with await, otherwise we may have strange behavior (finishing after main process finish).

Also applies the same comment of previous method without return

Reply:
this is actually intention. notification can be "trigger and forget", are optional, and running them in parrallel help having a faster api response